pkg/controller/atlascustomrole/custom_role.go

```diff
@@ -0,0 +1,105 @@
+package atlascustomrole
+
+import (
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translation/customroles"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+type roleController struct {
+	ctx       *workflow.Context
+	projectID string
+	service   customroles.CustomRoleService
+	role      *akov2.AtlasCustomRole
+	deleted   bool
+}
+
+func handleCustomRole(ctx *workflow.Context, akoCustomRole *akov2.AtlasCustomRole) workflow.Result {
+	ctx.Log.Debug("starting custom role processing")
+	defer ctx.Log.Debug("finished custom role processing")
+
+	r := roleController{
+		ctx:       ctx,
+		service:   customroles.NewCustomRoles(ctx.SdkClient.CustomDatabaseRolesApi),
+		projectID: akoCustomRole.Spec.ProjectIDRef.ID,
+		deleted:   !akoCustomRole.DeletionTimestamp.IsZero(),
```

Comment:
stylistic nit: to reduce the amount of fields, I suggest to just inject `akoCustomRole` and derive `projectID` and `deleted` inside the `roleController`. Technically, there is no need to dependency-inject those and makes the struct unnecessary bigger.